index.js

```diff
@@ -142,82 +141,209 @@ var webProgressListener = {
   if (!geckoThread || !geckoThread.activity.counts) {
-    console.warn("Lolwhut? No Gecko thread? No hangs?");
-    return {numHangs: null, minBucketLowerBound: 0};
+    panel.port.emit("warning", "unavailableBHR");
+    return new Promise((resolve) => { resolve({numHangs: null, minBucketLowerBound: 0}); });
   }
-  // see the NOTE in mostRecentHangs() for caveats when using the activity.counts histogram
-  // to summarize, the ranges are the inclusive upper bound of the histogram rather than the inclusive lower bound
-  let numHangs = 0;
-  let minBucketLowerBound = Infinity;
-  geckoThread.activity.counts.forEach((count, i) => {
-    var lowerBound = geckoThread.activity.ranges[i - 1] + 1;
-    if (lowerBound >= gHangThreshold) {
-      numHangs += count;
-      minBucketLowerBound = Math.min(minBucketLowerBound, lowerBound);
+
+  return new Promise((resolve) => {
+    if (includeChildHangs) {
+      getChildThreadHangs().then((hangs) => {
+        // accumulate all of the counts in the child processes with the counts in the parent process
+        let counts = geckoThread.activity.counts.slice(0);
+        hangs.forEach((threadHangStats) => {
+          let childGeckoThread = threadHangStats.find(thread => thread.name == "Gecko_Child");
+          if (childGeckoThread && childGeckoThread.activity.counts) {
+            childGeckoThread.activity.counts.forEach((count, i) => { counts[i] += count; });
+          }
+        });
+        resolve(counts);
+      });
+    } else { // Just use the parent process stats
+      resolve(geckoThread.activity.counts);
     }
+  }).then((counts) => {
+    // see the NOTE in mostRecentHangs() for caveats when using the activity.counts histogram
+    // to summarize, the ranges are the inclusive upper bound of the histogram rather than the inclusive lower bound
+    let numHangs = 0;
+    let minBucketLowerBound = Infinity;
+    counts.forEach((count, i) => {
+      var lowerBound = geckoThread.activity.ranges[i - 1] + 1;
+      if (lowerBound >= gHangThreshold) {
+        numHangs += count;
+        minBucketLowerBound = Math.min(minBucketLowerBound, lowerBound);
+      }
+    });
+    panel.port.emit("warning", null);
+    return {numHangs: numHangs, minBucketLowerBound: minBucketLowerBound};
   });
-  return {numHangs: numHangs, minBucketLowerBound: minBucketLowerBound};
 }
 
 function numEventLoopLags() {
-  try {
-    var snapshot = Services.telemetry.getHistogramById("EVENTLOOP_UI_ACTIVITY_EXP_MS").snapshot();
-  } catch (e) { // histogram doesn't exist, the Firefox version is likely older than 45.0a1
-    return {numHangs: null, minBucketLowerBound: 0};
-  }
-  let numHangs = 0;
-  let minBucketLowerBound = Infinity;
-  for (let i = 0; i < snapshot.ranges.length; ++i) {
-    if (snapshot.ranges[i] >= gHangThreshold) {
-      numHangs += snapshot.counts[i];
-      minBucketLowerBound = Math.min(minBucketLowerBound, snapshot.ranges[i]);
+  return new Promise((resolve) => {
```

Comment:
There's no asynchronous work here, so returning a Promise is just API, right? In which case you can just let the code stay the same and `return Promise.resolve(...`